### PR TITLE
Add spruce support

### PR DESCRIPTION
**Pull request by allomov**
This PR contains everything is needed to use spruce with bosh-workspace. By default `spiff` is used, but you can tell to use spruce in your deployment in the following way:
```
merge_tool: spruce
```
There is also an option to freeze manifest version in the following way:
```
merge_tool:
  name: spruce
  version: 0.9.0
```
In this case if someone use version of `spruce` that differs from `0.9.0`, he will get a warning notification about wrong version.

I have done this changes because it is impossible to spiff cf-release templates for the latest versions of BOSH that started to use `cloud-config` feature as the only way to setup infrastructure settings.

### CODE REVIEW

File: lib/bosh/workspace/merge_tool.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+module Bosh::Workspace
+  class MergeTool
+    include Bosh::Exec
+
+    attr_accessor :name, :version
+
+    def initialize(merge_tool = nil)
+      @name, @version = case merge_tool
+      when Hash
```
Comment (rkoster) [inline lib/bosh/workspace/merge_tool.rb]: Could you move the schema validations into [schemas/project_deployment.rb](https://github.com/cloudfoundry-incubator/bosh-workspace/blob/master/lib/bosh/workspace/schemas/project_deployment.rb)?

File: lib/bosh/workspace/merge_tool.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        command_failed(result.command, result.output) if result.failed?
+        yield result.output
+      end
+    end
+
+    def check_tool_version
+      run_merge_tool('-v') do |output|
+        actual_version = output.match(/(\d+\.\d+\.\d+)/).to_a.first
+        if actual_version.nil? || actual_version != version
+          say "[WARNING] Deployment requires #{name} to have version #{version}\n".make_yellow
```
Comment (rkoster) [inline lib/bosh/workspace/merge_tool.rb]: If I'm not mistaken you can use the [`warning`](https://github.com/cloudfoundry/bosh/blob/master/bosh_cli/lib/cli/core_ext.rb#L111) method here.

**Comment by rkoster** [comment]
@allomov really nice work, I have added some comments above. Looking forward to merging this PR :smile:

File: lib/bosh/workspace/merge_tool.rb (hunk 2 above)
Comment (allomov) [inline lib/bosh/workspace/merge_tool.rb]: You are right. Fixed.

File: lib/bosh/workspace/merge_tool.rb (hunk 1 above)
Comment (allomov) [inline lib/bosh/workspace/merge_tool.rb]: Sure. Done.

**Comment by rkoster** [comment]
@allomov thanks

**Comment by allomov** [comment]
Nice. Thank you for fast upstreaming of this changes.

**Pull request merged**